### ISSUE TITLE
question:  when "snapshot_create onchange" are 3 weekly snapshots, or 3 weeks of snapshots kept?

### ISSUE THREAD

**Issue by 1n5aN1aC** [digint/btrbk#365]
Good morning!  Great bit of software you have here, thanks for developing this!  I have a bit of a question on how backup retention works, and while I have a hunch what the answer is, I couldn't find a super-clear answer to, and don't want to make false assumptions.

Lets say I have `snapshot_create           onchange` defined in my config file, and a simplified config of `snapshot_preserve          7d 16w`

If 16 weeks of snapshots are taken, then no snapshots are taken for 10 weeks, (because no change were made, and snapshots were not made because of onchange) what is kept?

A.)  Are snapshots from the past 16weeks kept?  (This would mean ~6 snapshots are kept in the example above.)  If no changes were made for 16weeks, only 1 snapshot would be kept.

B.)  Are 16 weekly snapshots kept?  (This would mean 16 snapshots are kept in the example above- from the past ~25ish weeks)  If no changes are made for 16 weeks, the last 16 weekly snapshots will be pet.

**Comment by digint** [comment]
The answer is A.

btrbk.conf(5) states:
> weekly: Defines how many weeks back weekly backups should be preserved.

I'm not native english speaker, but "weeks back" should imply "weeks back from now". Was this your confusion? Any ideas how to reword this to make it more clear?

btw. using `btrbk run -n --print-schedule` gives a nice comprehensive output of what and why subvolumes get deleted.

**Comment by 1n5aN1aC** [comment]
Thanks, that answers my question very clearly, and that was my confusion.

I was going to answer how I got that confusion, but now I'm not sure.  I went back and looked at all the documentation, and it seems relatively clear on that point.  Maybe I just wasn't thinking clearly the day I tried to research it?

At any rate, thanks much!
